hedera-node/hedera-mono-service/src/main/java/com/hedera/node/app/spi/TransactionMetadata.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (C) 2022 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hedera.node.app.spi;
+
+import com.hedera.services.legacy.core.jproto.JKey;
+import com.hederahashgraph.api.proto.java.Transaction;
+import java.util.Collections;
+import java.util.List;
+
+/**
+ * Metadata collected when transactions are handled as part of "pre-handle". This happens with
+ * multiple background threads. Any state read or computed as part of this pre-handle, including any
+ * errors, are captured in the TransactionMetadata. This is then made available to the transaction
+ * during the "handle" phase as part of the HandleContext.
+ */
+public class TransactionMetadata {
+    private final boolean failed;
+    private final Transaction tx;
+    private final JKey payerSig;
```

Comment:
Since _every_ operation-specific method in a `PreTransactionHandler` will need to try to look up the payer key (and set the failure code in the `status` as appropriate), I suspect we need something like an,
```
abstract class PayerKeyTransactionMetadata {
    protected @Nullable JKey payerKey;
    protected ResponseCodeEnum status = OK;

    // No default constructor to force subclasses to give us a chance to resolve payer signing metadata
    protected PayerKeyTransactionMetadata(States states) {
        // Try to look up the payer key, set status to failure code if appropriate
    }
}
```

Reply:
Added `TransactionMetadata` interface and two implementations `SigTransactionMetadata` and `invalidTransactionMetadata`. I thought we could add `List<TransactionSignatures>` to `SigTransactionMetadata` when using in `SigRequirements` class.